src/import.php

```diff
@@ -37,6 +37,346 @@
 set_time_limit(0); // Disable time limit, parsing may take a long time.
 session_write_close(); // Also don't care about the session (in fact, it locks the whole LOVD while this page is running).
 
+
+
+
+
+if (ACTION == 'schedule' && PATH_COUNT == 1) {
+    // Schedule files for import.
+    define('LOG_EVENT', 'ImportSchedule');
+    define('PAGE_TITLE', 'Schedule files for import');
+    $_T->printHeader();
+    $_T->printTitle();
+
+    // Require manager clearance.
+    lovd_requireAUTH(LEVEL_MANAGER);
+
+    // This feature requires you to have the data files and the archive settings configured.
+    if (empty($_INI['paths']['data_files']) || empty($_INI['paths']['data_files_archive'])) {
+        lovd_showInfoTable('To use this feature, you will need to configure both the "data_files" and "data_files_archive" paths.', 'stop');
+        $_T->printFooter();
+        exit;
+    }
+
+    // Loop through the files in the dir and find importable files.
+    $h = @opendir($_INI['paths']['data_files']);
+    if (!$h) {
+        lovd_showInfoTable('Can\'t open directory: ' . $_INI['paths']['data_files'], 'stop');
+        $_T->printFooter();
+        exit;
+    }
+
+    // Fetch all info from the database, for annotation and error reporting.
+    // This will group by filename.
+    $zScheduledFiles = $_DB->query('SELECT sf.*, u.name AS scheduled_by_name
+                                    FROM ' . TABLE_SCHEDULED_IMPORTS . ' AS sf LEFT OUTER JOIN ' . TABLE_USERS . ' AS u ON (sf.scheduled_by = u.id)
+                                    ORDER BY (sf.processed_date IS NOT NULL) DESC, sf.priority DESC, sf.processed_date, sf.scheduled_date, sf.filename')
+                           ->fetchAllGroupAssoc();
+
+    // [0/1] indicates processed no/yes.
+    // Files in format [filename] = sort_code (unscheduled(0|1), reverse_priority(0-9), processed_date, scheduled_date, file_timestamp).
+    // NOTE: If you change the sort code, be sure to update this comment, the code that builds it, the code the fills in the orphaned database entries, and the scheduling part.
+    $aFiles = array(
+        array(),
+        array(),
+    );
```

Comment:
I had trouble understanding this format. If I understand correctly, $aFiles is an array where the key is a boolean denoting whether the files listed in the value array are processed or not. And the value itself is an array with filename as key and the value is a sort key in the form of a comma-separated string of the values:
* unscheduled (bool)
* reverse priority (integer between 0 and 9)
* date when processed
* date when scheduled
* date when file was created.

Perhaps an example is more telling:
~~~php
$aFiles = array(
    0 => array(
        'file.txt' => '0,4,2017-01-01 15:10:35,2017-01-01 15:10:36,2017-01-01 15:10:37',
        'file2.txt' => ...
    ),
    1 => ...
);
~~~
But maybe it's better to do away with the sort string altogether (see comment below).

Reply:
Yes, you understood correctly but you're right this code is complicated. It took me quite some time to build as well :stuck_out_tongue_closed_eyes: 
More on the other comment...